packages/react-liveness/src/components/FaceLivenessDetector/service/machine/machine.ts

```diff
@@ -98,17 +99,25 @@ const responseStreamActor = async (callback: StreamActorCallback) => {
     }
   } catch (error: unknown) {
     let returnedError = error;
+
     if (isInvalidSignatureRegionException(error)) {
       returnedError = new Error(
         'Invalid region in FaceLivenessDetector or credentials are scoped to the wrong region.'
       );
     }
 
     if (returnedError instanceof Error) {
-      callback({
-        type: 'SERVER_ERROR',
-        data: { error: returnedError },
-      });
+      if (isConnectionTimeoutError(error)) {
+        callback({
+          type: 'CONNECTION_TIMEOUT',
+          data: { error: returnedError },
+        });
+      } else {
+        callback({
+          type: 'SERVER_ERROR',
+          data: { error: returnedError },
+        });
+      }
```

Comment:
Think the logic in this `catch` could be simplified a bit for future readability:
```suggestion
    if (isInvalidSignatureRegionException(error)) {
      callback({ type: 'SERVER_ERROR', error: new Error(
        'Invalid region in FaceLivenessDetector or credentials are scoped to the wrong region.'
      ) })
    } else if (error instanceof Error) {
      callback({
        type: isConnectionTimeoutError(error) ? 'CONNECTION_TIMEOUT' : 'SERVER_ERROR',
        data: { error },
      });
    }
```